cg/services/order_validation_service/workflows/tomte/validation/inter_field/pedigree/models.py

```diff
@@ -0,0 +1,84 @@
+from cg.services.order_validation_service.models.errors import (
+    DescendantAsFatherError,
+    DescendantAsMotherError,
+    PedigreeError,
+    SampleIsOwnFatherError,
+    SampleIsOwnMotherError,
+)
+from cg.services.order_validation_service.workflows.tomte.models.case import TomteCase
+from cg.services.order_validation_service.workflows.tomte.models.sample import TomteSample
+
+
+class Node:
+    def __init__(self, sample: TomteSample):
+        self.sample = sample
+        self.father: Node | None = None
+        self.mother: Node | None = None
+        self.visited = False
+        self.in_current_path = False
+
+
+class Pedigree:
+    def __init__(self, case: TomteCase):
+        self.graph: dict[str, Node] = {}
+        self.case = case
+        self._add_nodes()
+        self._add_parents()
+
+    def _add_nodes(self) -> None:
+        for sample in self.case.samples:
+            node = Node(sample=sample)
+            self.graph[sample.name] = node
+
+    def _add_parents(self) -> None:
+        for node in self.graph.values():
```

Comment:
It could be argued that it is confusing to use graph terms such as nodes in one case, but instead of calling the edges "edges" we call them parents. I am fine with having it like this, but you make the call.